src/osmosis/vault_robot.rs

```diff
@@ -71,51 +71,41 @@ impl<'a, R: Runner<'a>> OsmosisVaultRobot<'a, R> {
         force_withdraw_admin: &SigningAccount,
         treasury: &SigningAccount,
         base_pool: OsmosisTestPool,
-        reward_token_denoms: &[String],
-        reward1_pool_liquidity: Vec<Coin>,
-        reward2_pool_liquidity: Option<Vec<Coin>>,
+        reward1_pool: OsmosisTestPool,
+        reward2_pool: Option<OsmosisTestPool>,
         reward_liquidation_target: String,
         performance_fee: Decimal,
         test_config_path: &str,
     ) -> Self {
-        let gamm = Gamm::new(app);
         let api = MockApi::default();
 
         let test_config = TestConfig::from_yaml(test_config_path);
 
-        println!("base_pool_liquidity: {:?}", base_pool.liquidity);
-
         // Create base pool (the pool this vault will compound)
         let base_pool_id =
             create_osmosis_pool(app, &base_pool.pool_type, &base_pool.liquidity, admin);
-        println!("Pool ID: {}", base_pool_id);
         let base_pool = OsmosisPool::unchecked(base_pool_id);
 
         // Create pool for first reward token
-        let reward1_pool_id = gamm
-            .create_basic_pool(&reward1_pool_liquidity, admin)
-            .unwrap()
-            .data
-            .pool_id;
-        let reward1_pool = OsmosisPool::unchecked(reward1_pool_id);
-        let reward1_token = reward1_pool_liquidity
+        let reward1_pool_id =
+            create_osmosis_pool(app, &reward1_pool.pool_type, &reward1_pool.liquidity, admin);
```

Comment:
i think it would be cleaner to make this function into an impl on `OsmosisTestPool` so you can call `test_pool.create(app, admin)`

Reply:
True

Reply:
Fixed